### PR TITLE
azstart example for getting started guide on ms learn

### PR DESCRIPTION
**Pull request by sandeep-sen**
This sample will be used in the ms learn site, in the getting started section of Azure SDK for Go

### CODE REVIEW

File: sdk/azstart/go.mod [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module azstart
+
+go 1.19
```
Comment (jhendrixMSFT) [inline sdk/azstart/go.mod:3]: For consistency with our SDKs, this should be 1.18

File: sdk/azstart/azstart.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	// Construct a credential type from the azidentity package
+	// or the module defining the client you wish to use
+	credential, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// Construct an ARM client passing subscription ID, credential, & optional options
+	// This example creates a ResourceGroupsClient, but you can create any ARM client
+	client, err := armresources.NewResourceGroupsClient("<SubscriptionId>", credential, nil)
```
Comment (jhendrixMSFT) [inline sdk/azstart/azstart.go:27]: We're introducing "client factories" in ARM modules.  Unsure if it's been merged for `armresources` yet, but perhaps we want to wait for that in order to promote it?

Comment (tadelesh) [reply sdk/azstart/azstart.go:27]: Thanks @jhendrixMSFT for involving me. We are doing the refresh work to add client factory for all arm modules recently and will finish before the end of March. @sandeep-sen Could you wait for a while? I'll help to change the code after the new pattern could be adapted to all arm modules.

File: sdk/azstart/go.mod (hunk 1 above)
Comment (sandeep-sen) [reply sdk/azstart/go.mod]: downgraded to 1.18

File: sdk/azstart/azstart.go (hunk 2 above)
Comment (sandeep-sen) [reply sdk/azstart/azstart.go:27]: @tadelesh I will wait till the client factory is released before I publish these code samples. I would need your assistance for updating this sample file

File: sdk/azstart/azstart.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+		azsecrets.SetSecretParameters{Value: to.Ptr("<SecretValue>")}, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// Use the service's response as your application desires
+	fmt.Printf("Name: %s, Value: %s\n", *response.ID, *response.Value)
+}
+
+// PagingOverACollection shows how to page over a collection's items
```
Comment (JeffreyRichter) [inline sdk/azstart/azstart.go:72]:
```suggestion
// ExamplePagingOverACollection shows how to page over a collection's items
```

File: sdk/azstart/azstart.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+		// Other clients may be adding/deleting items from the collection while
+		// this code is paging; some items may be skipped or returned multiple times.
+		for _, item := range page.Value {
+			_ = item // Here's where your code processes the item as you desire
+		}
+		// Looping around will request the next page of items from the service
+	}
+}
+
+// LongRunningOperation shows how to invoke a long-running operation and poll for its completion
```
Comment (JeffreyRichter) [inline sdk/azstart/azstart.go:105]:
```suggestion
// ExampleLongRunningOperation shows how to invoke a long-running operation and poll for its completion
```

**Review by JeffreyRichter — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**